.github/scripts/compile.sh

```diff
@@ -106,33 +106,49 @@ do_build () {
   local suffix=${1}
   local runtime_check=${2}
 
+  if [ "$runtime_check" == "OFF" ]; then
+     # Turn on PGO only when we are not doing asan or tsan
+     BUILD_OPTS="-DENABLE_PROFILE_GUIDED_OPTIMIZATION=ON"
+  else
+     BUILD_OPTS="-DBUILD_TESTING=OFF"
+  fi
+
   cmake -S "${PROTON_DIR}" -B "${PROTON_BUILD_DIR}${suffix}" \
     -DCMAKE_BUILD_TYPE=RelWithDebInfo \
     -DRUNTIME_CHECK="${runtime_check}" \
     -DENABLE_LINKTIME_OPTIMIZATION=ON \
-    -DCMAKE_POLICY_DEFAULT_CMP0069=NEW -DCMAKE_INTERPROCEDURAL_OPTIMIZATION=ON \
+    -DCMAKE_INTERPROCEDURAL_OPTIMIZATION=ON \
```

Comment:
Does this change still work to produce LTO build of Proton? Looking at the build log, I'm surprised that it actually does, https://github.com/skupperproject/skupper-router/actions/runs/4599414496/jobs/8124756301?pr=1044#step:3:1666

```
[  1%] Building C object c/CMakeFiles/qpid-proton-tls-static.dir/src/tls/openssl.c.o
cd /build/proton/app/build/c && /usr/bin/gcc -DPROTON_DECLARE_STATIC -I/build/proton/app/c/include -I/build/proton/app/c/src -I/build/proton/app/build/c/include -I/build/proton/app/build/c/src -O2 -flto=auto -ffat-lto-objects -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong   -m64 -march=x86-64-v2 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -fvisibility=hidden -O2 -g -DNDEBUG -flto -fno-fat-lto-objects -std=c99 -Werror -Wall -pedantic-errors -Wno-unused-parameter -Wstrict-prototypes -Wvla -Wsign-compare -Wwrite-strings  -MD -MT c/CMakeFiles/qpid-proton-tls-static.dir/src/tls/openssl.c.o -MF CMakeFiles/qpid-proton-tls-static.dir/src/tls/openssl.c.o.d -o CMakeFiles/qpid-proton-tls-static.dir/src/tls/openssl.c.o -c /build/proton/app/c/src/tls/openssl.c
```

I guess this is because minimal required cmake version in proton was increased and so this CMP0069 policy no longer applies.

Reply:
Note to /self: check that the proton static lib also gets lto flags

Reply:
Yes, it looks like `CMP0069` no longer applies. It showed up as an error when running cmake. The error went away after I removed `-DCMAKE_POLICY_DEFAULT_CMP0069=NEW`

Reply:
Even static lib gets now LTO. Ok, it really does work.